Framework/PythonInterface/plugins/algorithms/SetSampleFromLogs.py

```diff
@@ -0,0 +1,188 @@
+                if value == 1.0 or value == 0.0:
+                    material['Mass'] = _getLogValue(runObject, 'SampleMass')
+                elif _hasValue(runObject, "SampleMass"):
+                    material['SampleMassDensity'] = value
+
+        # log the results and return
+        self.log().information('MATERIAL: ' + str(material))
+        return material
+
+    def _createGeometry(self, runObject, instrEnum):
+        # get height of sample from the logs
+        # this assumes the shape has a "Height" property
+        geometry = _toDict(self.getProperty("Geometry").value)
+
+        if self.getProperty("FindGeometry").value:
+            if not _hasValue(geometry, "Height"):
+                heightInContainerNames = ['HeightInContainer']
+                heightInContainerUnitsNames = ['HeightInContainerUnits']
+
+                # determine "special" logs for SNS instruments
+                if instrEnum.facility().name() == 'SNS':
+                    beamline = ''
+                    # TODO this would benefit from the beamline exposed through python
+                    if instrEnum.name() == 'NOMAD':
+                        beamline = 'BL1B'
+                    elif instrEnum.name() == 'POWGEN':
+                        beamline = 'BL11A'
+                    else:
+                        warningMsg = 'Do not know how to create lognames for "{}"'.format(instrEnum.name())
+                        self.log().warn(warningMsg)
+                    if beamline:
+                        # "internal" log names at SNS are templated from the beamline number
+                        heightInContainerUnitsNames.append('{}:CS:ITEMS:HeightInContainerUnits'.format(beamline))
+                        heightInContainerNames.append('{}:CS:ITEMS:HeightInContainer'.format(beamline))
+                self.log().information("Looking for sample height from {} and units from {}".format(heightInContainerNames,
+                                                                                                    heightInContainerUnitsNames))
+                # Check units - SetSample expects cm
+                unitsKey = _findKey(runObject, *heightInContainerUnitsNames)
+                units = ''
+                if unitsKey:
+                    units = _getLogValue(runObject, unitsKey)
+                    # not finding units will generate a warning below
+                # create conversion factor into cm
+                conversion = 1.0  # don't do any conversion
+                if units == "cm":
+                    conversion = 1.0
+                elif units == "mm":
+                    conversion = 0.1
+                else:
+                    warningMsg = "HeightInContainerUnits expects cm or mm;" + \
+                                " specified units not recognized: {:s};".format(units) + \
+                                " we will reply on user input for sample density information."
+                    self.log().warn(warningMsg)
```

Comment:
Hopefully the logger writer at the DAS end is case insensitive so that we don't have to worry about having "Cm" or "MM" pops up